M2/Macaulay2/m2/html.m2

```diff
@@ -462,26 +462,80 @@ describeReturnCode = r -> (
      rundir := temporaryFileName() | "-rundir/";
-     cmd := ulimit | "cd " | rundir | "; " | env | cmdname | " " | args | " <" | format inf | " >>" | format toAbsolutePath tmpf | " 2>&1";
-     stderr << cmd << endl;
      makeDirectory rundir;
+     -* by default, the following commandline fixtures are used *-
+     defmode := (SetUlimit + GCMAXHEAP + ArgQ + ArgInt
+	 + ArgNoRandomize + ArgNoReadline + ArgSilent + ArgStop + ArgPrintWidth
+	 + SetInputFile + SetOutputFile + SetCaptureErr);
+     -* convert usermode to a number and use the bitmask above to decide arguments *-
+     usermode = if usermode === true then defmode - 1 else usermode; -* the default without -q *-
+     usermode = if usermode === false or usermode === null then defmode else usermode; -* the default *-
+     -* as a convenience, to start with the default and add or subtract args,
+        negate the integer by subtracting 2^64; here we undo those actions: *-
+     usermode = if usermode < 0 then (1<<64) + defmode - usermode else usermode;
+     -* returns (" "|arg) if all bits in m are set in usermode *-
+     readmode := (m, arg) -> if usermode & m == m then " " | arg else "";
+     cmd := readmode(SetUlimit, ulimit);
+     cmd = cmd | " cd " | rundir | ";";
+     cmd = cmd | readmode(GCMAXHEAP,      "GC_MAXIMUM_HEAP_SIZE=400M");
+     cmd = cmd | readmode(GCSTATS,        "GC_PRINT_STATS=1");
+     cmd = cmd | readmode(GCVERBOSE,      "GC_PRINT_VERBOSE_STATS=1");
+     cmd = cmd | " " | format toAbsolutePath commandLine#0;
+     cmd = cmd | readmode(ArgQ,           "-q");
+     cmd = cmd | readmode(ArgInt,         "--int");
+     cmd = cmd | readmode(ArgNoBacktrace, "--no-backtrace");
+     cmd = cmd | readmode(ArgNoDebug,     "--no-debug");
+     cmd = cmd | readmode(ArgNoPreload,   "--no-preload");
+     cmd = cmd | readmode(ArgNoRandomize, "--no-randomize");
+     cmd = cmd | readmode(ArgNoReadline,  "--no-readline");
+     cmd = cmd | readmode(ArgNoSetup,     "--no-setup");
+     cmd = cmd | readmode(ArgNoThreads,   "--no-threads");
+     cmd = cmd | readmode(ArgNoTTY,       "--no-tty");
+     cmd = cmd | readmode(ArgNoTValues,   "--no-tvalues");
+     cmd = cmd | readmode(ArgNotify,      "--notify");
+     cmd = cmd | readmode(ArgSilent,      "--silent");
+     cmd = cmd | readmode(ArgStop,        "--stop");
+     cmd = cmd | readmode(ArgPrintWidth,  "--print-width 77");
+     cmd = cmd | concatenate apply(srcdirs, d -> (" --srcdir",format d));
+     needsline := concatenate(" -e 'needsPackage(\"",pkgname,"\", Reload => true, FileName => \"",pkg#"source file","\")'");
+     cmd = cmd | if pkgname != "Macaulay2Doc" then needsline else "";
+     cmd = cmd | readmode(SetInputFile,   "<" | format inf);
+     cmd = cmd | readmode(SetOutputFile,  ">>" | format toAbsolutePath tmpf);
+     cmd = cmd | readmode(SetCaptureErr,  "2>&1");
+     if debugLevel > 0 then stderr << cmd << endl;
```

Comment:
Sometimes the user will need the exact command line for debugging purposes, but now it will be hidden, with no documented way, aside from reading the source code, for getting it to be displayed.   If you really want this useful information to be hidden, then it should probably be connected to the Verbose option and documented.

Reply:
set `debugLevel = 1`.

Reply:
As you said:
> Sometimes the user will need the exact command line for **debugging purposes**

So it is clear.

Reply:
Well, sure, if the user reads the source code, then they can figure it out.  But what user will be able to do that?  That's a high bar for users.

Reply:
Then that's a problem with the Core that sound be fixed. Increasing `debugLevel` to get debug output is a natural thing to do.

Reply:
That idea might not occur to all users, if it's not documented.  Here's an idea: display the command afterwards, if the example or test failed.

Reply:
Done.